swagger.yaml

```diff
@@ -2500,6 +2500,105 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /clusters/{cluster_id}/operators:
+    get:
+      tags:
+        - operators
+        - installer
+      security:
+        - agentAuth: []
+      description: Lists operators for a cluster.
+      operationId: ListOfClusterOperators
+      parameters:
+        - in: path
+          name: cluster_id
+          description: The cluster to return operators for.
+          type: string
+          format: uuid
+          required: true
+        - in: query
+          name: operator_name
+          description: An operator in the specified cluster to return its data.
+          type: string
+          required: false
+      responses:
+        "201":
+          description: Success.
+        "401":
+          description: Unauthorized.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "403":
+          description: Forbidden.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "404":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "405":
+          description: Method Not Allowed.
+          schema:
+            $ref: '#/definitions/error'
+        "500":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+    post:
+      tags:
+        - operators
+        - installer
+      security:
+        - agentAuth: []
+      description: Controller API to report of monitored operators.
```

Comment:
This API should be used by both the service and the controller, right ?
The service will define the list of operators that the controller needs to monitor.
In this context it seems that only the custom/olm operators should be added by the service.
For OLM operators the name and namespace are required to exist prior the controller so it can know how to monitor them.

When the controller runs, it will report the status of those custom/olm operators back by updating the list.
In addition the controller will add to the list all of the cluster-operators + cvo that don't require any input from the service side on how to monitor them.

Does it make sense ?
If that makes sense, perhaps it would be better to split into two collections:
```
/clusters/{cluster_id}/operators

This will include the operators created by the assisted-service (per user request) to be monitored by the controller and how.
The operator will include name, namespace and timeout.
```
and
```
/clusters/{cluster_id}/monitored_operators

This will include the reported list of operators by the controller:
- the OLM operators (based on the previous collection)
- the cluster operators
- the CVO

The `monitored_operator` will contain the status, status_info, updated_at
```

Reply:
@masayag I don't understand the comment - what does it mean for an API to be used by the service?  The service doesn't call itself.  Can you please explain?

Reply:
@avishayt My intention was that the from UX perspective, the user will be able to mark OCS, LSO, CNV as operators to deploy on the cluster.
Those options (of which operator to deploy) will be set on the cluster by PUT requests to `/clusters/{cluster_id}/operators` 
I referred to the UI as part of the service, so let me correct myself and use "clients of the service".

Reply:
I understand now, thanks.  I assumed this API was only for the agent to report progress, and that the UI/client would use PATCH on the cluster resource to select the operators to deploy and with what options.  I guess this is the custom_operators property on the cluster, but not sure what it means.